python/ray/data/preprocessor.py

```diff
@@ -64,7 +64,7 @@ class FitStatus(str, Enum):
     def fit_status(self) -> "Preprocessor.FitStatus":
         if not self._is_fittable:
             return Preprocessor.FitStatus.NOT_FITTABLE
-        elif self._check_is_fitted():
+        elif hasattr(self, "_fitted") and self._fitted:
```

Comment:
Nit: Rather than checking `hasattr(self, "_fitted")`, would it make sense to set `self._fitted = False` in the constructor? Initializing attributes in the constructor is a good practice.

https://pylint.pycqa.org/en/latest/user_guide/messages/warning/attribute-defined-outside-init.html

Reply:
It's kind of tricky here since this is an abstract base class. The alternative is we define an abstract `__init__` method, and then all subclasses need to call `super().__init__()`